applications/GeoMechanicsApplication/custom_constitutive/thermal_dispersion_2D_law.cpp

```diff
@@ -0,0 +1,67 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Mohamed Nabi
+//                   John van Esch
+//
+
+#include "custom_constitutive/thermal_dispersion_2D_law.hpp"
+#include "geo_mechanics_application_variables.h"
+#include <iostream>
+
+namespace Kratos {
+
+ConstitutiveLaw::Pointer GeoThermalDispersion2DLaw::Clone() const
```

Comment:
There is some discussion with @mnabideltares @WPK4FEM @avdg81 about the naming of this class. Since this law is about the heat transfer, which at this point is only conduction, but will be extended with convection later on, I would say a more descriptive name would be e.g. GeoThermalTransferLaw, or GeoThermalConductionAndConvectionLaw if we really want to be specific. Let's discuss in office